### ISSUE TITLE
Salary slip and calculating the overtime based on the employee hour rate

### ISSUE THREAD

**Issue by bghayad** [frappe/erpnext#10386]
Hello;
It is impossible to have flat price for the hour rate for all employees. Logically, each employee has hour rate differs than other employee as his salary value is also different!
Considering the hour rate for the overtime to be fixed for all employees useless. The overtime hour rate should be based on formula that has the salary components (basic, home allowance, ... etc).
Or:
If possible to get the number of hours from the timesheet, so we can calculate the overtime (as salary component) based on formula, it is also fine. But as I see that the total hours variable can not be used in the salary structure.
Regards
Bilal

**Comment by umairsy** [comment]
In the Salary Structure, if you check field "Salary Slip Based on Timesheet", then you will find an option to select Earning Deduction component for it, with Hour Rate.

![screen shot 2017-08-16 at 2 36 28 pm](https://user-images.githubusercontent.com/910238/29356554-3e841bf6-8292-11e7-9658-651754d1fac7.png)

Based on this setup, you will only have overtime hours fetched from Timesheet. The value for the component of overtime will be calculated based on the total hours from Timesheet * Hour Rate mentioned in the Timesheet.

**Comment by bghayad** [comment]
I found a work around solution for this problem: The total_hours can be used to determine the total overtime hours fetched from the timesheet, so it is possible in this case to calculate the overtime based on the employee salary (the hour rate is B/(26 * 8) and the total_hours is known).
The only remaining problem now that the Overtime salary component that is selected when selecting salary slip based on timesheet is appearing in the salary slip (even if I placed the hour rate to be 0 as I do not care any more for the hour rate). So I would if it possible to disappear from the salary slip if I placed the hour rate 0.
